src/Open3D/GUI/Menu.cpp

```diff
@@ -69,59 +93,44 @@ void Menu::AddItem(const char *name,
 }
 
 void Menu::AddMenu(const char *name, std::shared_ptr<Menu> submenu) {
-    impl_->items.push_back({NO_ITEM, name, "", submenu});
+    impl_->items.push_back({NO_ITEM, name, "", submenu, submenu->impl_.get()});
 }
 
 void Menu::AddSeparator() {
-    impl_->items.push_back({NO_ITEM, "", "", nullptr, false, false, true});
+    impl_->items.push_back(
+            {NO_ITEM, "", "", nullptr, nullptr, false, false, true});
```

Comment:
Maybe introduce static function for creating separator? Something like `MenuItem::Separator()`, all this nullptrs and flags just blow my mind %)

Reply:
This *is* the function for creating the separator, I'm not sure what passing it off to another function that does exactly the same would acheive...